migrator/migrator.go

```diff
@@ -0,0 +1,135 @@
+package migrator
+
+import (
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"io/ioutil"
+
+	"github.com/pkg/errors"
+	"golang.org/x/tools/go/ast/astutil"
+)
+
+type Walker struct {
+	fset *token.FileSet
+	f    *ast.File
+
+	packageNameAE          string
+	packageNameAEDatastore string
+	packageNameGoon        string
+	packageNameBoom        string
+	packageContext         string
+	useIterator            bool
+
+	clientVarName  string
+	contextVarName string
+	queryVarName   string
+	txVarName      string
+	commitVarName  string
+	goonVarName    string
+	boomVarName    string
+	goonTxName     string
+
+	hasAEDatastorePackage bool
+	hasGoonPackage        bool
+
+	willModify map[ast.Node]func(c *astutil.Cursor) bool
+}
+
+func Main(targetFiles []string) (*token.FileSet, []*ast.File, error) {
+	fset := token.NewFileSet()
+	fs := make([]*ast.File, 0, len(targetFiles))
+	for _, targetFile := range targetFiles {
+		f, err := MigrateFile(fset, targetFile)
+		if err != nil {
+			return nil, nil, err
+		}
+		fs = append(fs, f)
+	}
+
+	return fset, fs, nil
+}
+
+func MigrateFile(fset *token.FileSet, targetFile string) (*ast.File, error) {
+	// 指定されたファイルをASTに変換する
+	b, err := ioutil.ReadFile(targetFile)
+	if err != nil {
+		return nil, errors.Wrapf(err, "ioutil.ReadFile: %s", targetFile)
+	}
+
+	f, err := parser.ParseFile(fset, targetFile, b, parser.ParseComments)
```

Comment:
[IMO]
Third parameter of `parser.ParseFile` accepts `io.Reader` too.

Reply:
👀